ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx

```diff
@@ -133,23 +138,48 @@ export function PublishOnDemandWidget(props: PublishOnDemandWidgetProps) {
   const [initialPublishingTarget, setInitialPublishingTarget] = useState(null);
   const [publishingTargets, setPublishingTargets] = useState(null);
   const [publishingTargetsError, setPublishingTargetsError] = useState(null);
+  const { bulkPublishCommentRequired, publishByCommitCommentRequired, publishEverythingCommentRequired } = useSelection(
+    (state) => state.uiConfig.publishing
+  );
   const [publishGitFormData, setPublishGitFormData] = useSpreadState<PublishFormData>(initialPublishGitFormData);
-  const [publishGitFormValid, setPublishGitFormValid] = useState(false);
+  const publishGitFormValid =
+    publishGitFormData.commitIds.replace(/\s/g, '') !== '' &&
+    publishGitFormData.publishingTarget !== '' &&
+    (!publishByCommitCommentRequired || !isBlank(publishGitFormData.comment));
```

Comment:
Cheapest statements first
```suggestion
    publishGitFormData.publishingTarget !== '' &&
    (!publishByCommitCommentRequired || !isBlank(publishGitFormData.comment)) &&
    publishGitFormData.commitIds.replace(/\s/g, '') !== '';
```